src/main/java/uk/gov/digital/ho/hocs/casework/domain/repository/CaseDataRepository.java

```diff
@@ -25,4 +26,7 @@ public interface CaseDataRepository extends CrudRepository<CaseData, Long> {
     @Query(value = "SELECT cd.* FROM case_data cd WHERE cd.uuid = ?1", nativeQuery = true)
     CaseData findAnyByUuid(UUID uuid);
 
+    @Query(value="SELECT cast(cd.uuid as text) FROM case_data cd WHERE cd.migrated_reference = ?1 LIMIT 1", nativeQuery = true)
```

Comment:
Happy to be wrong but do we need to do this cast? I'd have thought/ hoped it could do this internally

Reply:
The postgres dialect can't convert from the raw UUID representation in the db, throws "No Dialect mapping for JDBC type: 1111" exception. (where 1111 is a generic "OTHER" type that applies to UUIDs, JSONB, etc.)

There might be some tech debt work here to setup the dialect so it does just work, but that seems outside the scope of this feature.